lightning/src/onion_message/messenger.rs

```diff
@@ -283,6 +283,36 @@ where
 		&self, path: OnionMessagePath, message: OnionMessageContents<T>,
 		reply_path: Option<BlindedPath>
 	) -> Result<(), SendError> {
+		let (introduction_node_id, onion_msg) = Self::make_onion_message(
+			&self.entropy_source, 
+			&self.node_signer, 
+			&self.secp_ctx, 
+			path, 
+			message, 
+			reply_path
+		)?;
+
+		let mut pending_per_peer_msgs = self.pending_messages.lock().unwrap();
+		if outbound_buffer_full(&introduction_node_id, &pending_per_peer_msgs) { return Err(SendError::BufferFull) }
+		match pending_per_peer_msgs.entry(introduction_node_id) {
+			hash_map::Entry::Vacant(_) => Err(SendError::InvalidFirstHop),
+			hash_map::Entry::Occupied(mut e) => {
+				e.get_mut().push_back(onion_msg);
+				Ok(())
+			}
+		}
+	}
+
+	/// Make an onion message with contents `message` to the destination of `path`.
+	/// Returns (introduction_node_id, onion_msg)
+	pub fn make_onion_message<T: CustomOnionMessageContents>(
```

Comment:
Nit: prefer `create_onion_message`, more consistent with other APIs

Reply:
ok, i renamed it to `create_onion_message`